pkg/controller/servicebindingrequest/binder_test.go

```diff
@@ -36,24 +38,82 @@ func getEnvVar(envVars []corev1.EnvVar, name string) *corev1.EnvVar {
 	return nil
 }
 
+func TestBindingCustomSecretPath(t *testing.T) {
+	ns := "custombinder"
+	name := "service-binding-request-custom"
+	matchLabels := map[string]string{
+		"appx": "x",
+	}
+
+	f := mocks.NewFake(t, ns)
+	sbrSecretPath := f.AddMockedServiceBindingRequest(name, &ns, "ref-custom-podspec", "deployment", deploymentsGVR, matchLabels)
+	f.AddMockedUnstructuredDeployment("deployment", matchLabels)
+
+	customSecretPath := "metadata.clusterName"
+	sbrSecretPath.Spec.ApplicationSelector.BindingPath = &v1alpha1.BindingPath{
+		SecretPath: customSecretPath,
+	}
+	binderForsbrSecretPath := newBinder(
+		context.TODO(),
+		f.FakeDynClient(),
+		sbrSecretPath,
+		[]string{},
+		testutils.BuildTestRESTMapper(),
+	)
+	require.NotNil(t, binderForsbrSecretPath)
+
+	t.Run("custom secret field path", func(t *testing.T) {
+		secretPath := binderForsbrSecretPath.getSecretFieldPath()
+		expectedSecretPath := []string{"metadata", "clusterName"}
+		require.Equal(t, expectedSecretPath, secretPath)
+	})
+
+	t.Run("update custom secret field path ", func(t *testing.T) {
+		list, err := binderForsbrSecretPath.search()
+		require.NoError(t, err)
+		require.Len(t, list.Items, 1)
+
+		obj := list.Items[0]
+		err = binderForsbrSecretPath.updateSecretField(&obj)
+		require.NoError(t, err)
+		customSecretPathSlice := strings.Split(customSecretPath, ".")
+
+		customSecretInMeta, found, err := unstructured.NestedFieldCopy(obj.Object, customSecretPathSlice...)
+		require.NoError(t, err)
+		require.True(t, found)
+		require.Equal(t, name, customSecretInMeta)
+	})
+}
+
 func TestBinderNew(t *testing.T) {
 	ns := "binder"
 	name := "service-binding-request"
 	matchLabels := map[string]string{
 		"connects-to": "database",
 		"environment": "binder",
 	}
+
 	f := mocks.NewFake(t, ns)
 	sbr := f.AddMockedServiceBindingRequest(name, nil, "ref", "", deploymentsGVR, matchLabels)
+	ensureDefaults(&sbr.Spec.ApplicationSelector)
```

Comment:
As a note, this line should be called somewhere in `AddMockedServiceBindingRequest`'s call tree.

This ensures every test using that method is being returned a valid and complete application selector in the resulting `sbr` without requiring the knowledge of calling `ensureDefaults` immediately after calling `AddMockedServiceBindingRequest`.

Reply:
The method `AddMockedServiceBindingRequest's` is in a separate package and `ensureDefaults` is not exported.
I hope it's fine to leave it like this.

Reply:
can we move `AddMockedServiceBindingRequest` to the same package where `ensureDefaults` is?

Reply:
If this is necessary, we can do it in another PR.